doc/module_examples/04_fix_issue/check.sh

```diff
@@ -0,0 +1,168 @@
+
+#
+# The path to the backed up config file
+#
+dst_foo_conf="${PREUPG_CLEANCONFDIR}${foo_conf}"
+
+#
+# The name of the post-upgrade script
+#
+post_script="04_fix_issue_postupgrade.sh"
+
+#
+# The expected exit code
+#
+ret=$RESULT_PASS
+
+#
+# Set the expected exit code according to the given parameter and current value
+# of the $ret variable.
+#
+# @param  exit code; expected values are: $RESULT_FAIL, $RESULT_FIXED
+#
+set_result() {
+  case $1 in
+    $RESULT_FAIL)  ret=$1 ;;
+    $RESULT_FIXED) [ $ret -ne $RESULT_FAIL ] && ret=$1 ;;
+  esac
+}
+
+
+###########################################################
+# MAIN                                                    #
+###########################################################
+
+if [[ ! -e "$foo_conf" ]]; then
+  # The file is required, so log "error" and exit with "error" when the file doesn't
+  # exist.
+  log_error "The $foo_conf file doesn't exist but it is required by the naughty-foo package."
+  exit_error
+fi
+
+# case 1)
+if grep -q "obsoleted_option" "$foo_conf"; then
+  log_info "The 'obsoleted_option' in '$foo_conf' has been renamed on the new system to 'new_option'."
+
+  # Store the output file in a special directory together with its parents.
+  # Use the $PREUPG_CLEANCONFDIR directory for this purpose (see above). However,
+  # check first that the file hasn't been backed up already.
+  # (Provided that everything else is compatible; otherwise do
+  #  not use that directory and use the dirtyconf directory instead.)
+  [ -e "$dst_foo_conf" ] \
+    || cp -ar "$foo_conf" "$PREUPG_CLEANCONFDIR"
+  sed -ir 's/^[[:space:]]*obsoleted_option([[:space:]]|$)/new_option /' \
+    > "$dst_foo_conf"
+  {
+    echo -n "The \"obsoleted_option\" in the $foo_conf file has been renamed"
+    echo -n " to \"new_option\" on the new system. This has been fixed"
+    echo -n " and the fixed configuration file will be applied on the new"
+    echo    " system automatically."
+    echo
+  } >> "$SOLUTION_FILE"
+  set_result $RESULT_FIXED
+fi
+
+# case 2)
+if grep -q "deprecated_option" "$foo_conf"; then
+  log_medium_risk "The 'deprecated_option' is used inside '$foo_conf'."
+
+  if [ -e "$dst_foo_conf" ]; then
+    || cp -ar "$foo_conf" "$PREUPG_CLEANCONFDIR"
+  sed -ir 's/^[[:space:]]*deprecated_option([[:space:]]|$)/#deprecated_option /' \
+    > "$dst_foo_conf"
+  {
+    echo -n "The \"deprecated_option\" option has been removed from the new"
+    echo -n " system. The option has been commented out. Check the correct"
+    echo -n " functionality of your applications."
+    echo
+  } >> "$SOLUTION_FILE"
+  set_result $RESULT_FAIL
+fi
+
+# case 3)
+# ok, let's pretend that when a naughty-foo-cottage subpackage is installed,
+# we will want to install a naughty-foo-house on the new system
+if is_pkg_installed "naughty-foo-house"; then
+  log_info "The 'naughty-foo-cottage' package has been split on the new system. A 'naughty-foo-house' package will be installed."
```

Comment:
Split is not the best word here IMO. First time I saw that I was like "What that is suppose to mean? Is it some kind of jargon I don't know about?" :) Better would be *".. has been separated into two different packages .."*. Or use *divided* instead of *separated*, if you wish.